packages/firestore/src/util/async_queue.ts

```diff
@@ -326,14 +326,14 @@ export class AsyncQueue {
       `Attempted to schedule multiple operations with timer id ${timerId}.`
     );
 
-    const delayedOp = DelayedOperation.createAndSchedule<unknown>(
+    const delayedOp = DelayedOperation.createAndSchedule<T>(
       this,
       timerId,
       delayMs,
       op,
-      op => this.removeDelayedOperation(op)
+      op => this.removeDelayedOperation(op as DelayedOperation<unknown>)
```

Comment:
Hmmm. This looks wrong to me. Not the code you wrote - but don't we need to call `removeDelayedOperation` and `delayedOperations.push` with delayedOp?

Reply:
I agree it looks wrong, but it's actually right because unlike the line above, the removal callback is called with the delayedOp. It's pretty confusing though since we call it "op" which is already an in-scope variable.  I renamed it to `removedOp` to avoid ambiguity.  I tried `removedDelayedOp` but that is a bit long and prettier ended up splitting `DelayedOperation<unknown>` into 3 lines which looked horrible.

Reply:
Oh. Yes, that is confusing. I even checked out the branch locally and it didn't clear up this confusion :/ Thanks for renaming.